src/modules/Popup/Popup.js

```diff
@@ -171,6 +177,12 @@ export default class Popup extends Component {
     return { contentRestProps, portalRestProps }
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.open && shallowEqual(this.props.popperDependencies, prevProps.popperDependencies)) {
```

Comment:
`shallowEqual` for dependency prop doesn't fit to hook-like dependency input.

React hook uses internal `areHookInputsEqual(nextDeps, prevDeps)` methods which iteratively check both input items using  [`Object.is()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) in order.

See: 
https://github.com/facebook/react/blob/45acbdc0ba05c29f849011eaed5e5b221b219d06/packages/react-reconciler/src/ReactFiberHooks.js#L301-L348

And there is a clone for above method:
https://github.com/Andarist/are-hook-inputs-equal/blob/master/src/index.js

Reply:
I will cover this with UTs to be sure 🐱